### ISSUE TITLE
Need to know which USB WiFi adapter uses Atheros AR9280 you recommend ?

### ISSUE THREAD

**Issue by bobwilmes** [seemoo-lab/owl#18]
Trying to find the WiFi USB adapter using Atheros AR9280 you recommend

**Comment by CaptainStabs** [comment]
The AWUSO36NHA works well for this (Artheros AR9271)

**Comment by treysis** [comment]
Doesn't do 5 GHz though.

**Comment by thomasst** [comment]
I just got the Alfa AWUS036NHA with AR9271 chipset (which supposedly supports packet injection).

```
# dmesg | grep ath
[    3.756113] usb 1-11: ath9k_htc: Firmware ath9k_htc/htc_9271-1.4.0.fw requested
[    3.756156] usbcore: registered new interface driver ath9k_htc
[    4.038715] usb 1-11: ath9k_htc: Transferred FW: ath9k_htc/htc_9271-1.4.0.fw, size: 51008
[    4.290554] ath9k_htc 1-11:1.0: ath9k_htc: HTC initialized with 33 credits
[    4.556824] ath9k_htc 1-11:1.0: ath9k_htc: FW Version: 1.4
[    4.556825] ath9k_htc 1-11:1.0: FW RMW support: On
[    4.556826] ath: EEPROM regdomain: 0x833a
[    4.556826] ath: EEPROM indicates we should expect a country code
[    4.556826] ath: doing EEPROM country->regdmn map search
[    4.556827] ath: country maps to regdmn code: 0x37
[    4.556827] ath: Country alpha2 being used: GB
[    4.556828] ath: Regpair used: 0x37
[    4.563618] ath9k_htc 1-11:1.0 wlx00c0ca981942: renamed from wlan0
# iw dev wlx00c0ca981942 info
Interface wlx00c0ca981942
	ifindex 5
	wdev 0x100000001
	addr 00:c0:ca:98:19:42
	type managed
	wiphy 1
	txpower 20.00 dBm
# iw phy phy1 info | grep supports
	Device supports RSN-IBSS.
	Device supports T-DLS.
	Device supports TX status socket option.
	Device supports HT-IBSS.
	Device supports SAE with AUTHENTICATE command
	Device supports low priority scan.
	Device supports scan flush.
	Device supports AP scan.
	Device supports per-vif TX power setting
	Driver supports full state transitions for AP/GO clients
	Driver supports a userspace MPM
	Device supports configuring vdev MAC-addr on create.
# owl -i wlx00c0ca981942
              .oOXWMMMMWXOx:
         .oOOOx:'''''''''''':OOOx:
      oXOo'      ........      ':OXx.
           .oOOO''''''''''OOOo.
        oXOo'                'oOO:
             :oOOOOXXXXOOOOo:.
          oXO:'            ':OXo
              .:xOXXXXXXOx:.
          .xXMMMMMMMMMMMMMMMMXx.
  'XWWWWWWMMMMMMMMMMMMMMMMMMMMMMWWWWWWX'
    oWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMWo
     OMMMMMMWWMMMMMMMMMMMMMMWWWMMMMMO
    OMMWx'      'xWMMMMWx'      'oXMMO
   :MW:            oMMx            'WM:
   XM'    .xOOo.    :o     .xOOo.    WX
   WX    :MMMMMX          :MMMMMX    xW
   XW    'WMMMMX   .xx.   'WMMMWX    XX
   'Wx    'xWMx'   OMMO    'xWMx'   xM'
    'XX:           'XX'           :XX'
      'xXOx:..................:xXWx'
         'xXMMMMMMMMMMMMMMMMMMWO'

            Open Wireless Link

            https://owlink.org

21:34:37 ERROR: Error while receiving via netlink: Operation not supported
21:34:37 ERROR: Could not put device in monitor mode: wlx00c0ca981942
21:34:37 ERROR: could not initialize core
```

@CaptainStabs Is it actually working for you?

@schmittner Does this mean the AR9271 is unsupported, or is there anything else I can try on this card?

**Comment by thomasst** [comment]
If I start `owl` with `-N`, `opendrop find` seems to list the devices, but the device doesn't show up on my Mac or iPhone in AirDrop even though I have `opendrop receive` running (the device itself also shows up in `opendrop find`).

**Comment by jackforfaltu** [comment]
@thomasst 

Did you figure out how to start OWL or OpenDrop with this WiFi card?

I have the same one and was curious on trying this, but it didn't work for me either.

**Comment by jackforfaltu** [comment]
@CaptainStabs 

Could you please share how you were able to get OWL running with the AWUSO36NHA?

**Comment by thomasst** [comment]
@jackforfaltu I believe AR9271 is by default ACKing incoming frames*, so running it with the `-N` flag would be appropriate. However, I couldn't get it to appear on an Apple device when in receive mode. I'm assuming this is because 5 GHz is not supported by the AR9271 chipset, and it appears that AirDrop doesn't work well (at all?), see discussions in  https://github.com/seemoo-lab/opendrop/issues/4 and https://github.com/seemoo-lab/opendrop/issues/21#issuecomment-531733558 regarding channels.

*According to https://github.com/rofl0r/wpakey (I bolded the relevant part):
> the only fix for this issue is the so-called "active monitor" mode. currently, this feature can only be activated on ath9k and mt7601u drivers using iw dev wlan1 set monitor active. iw list or iw phyX show can tell you whether the feature is implemented, it will print Device supports active monitor (which will ACK incoming frames) if supported. **note that even though this feature cannot be actively activated on ath9k_htc devices, some or all of them have this behaviour turned on by default, so it may well be that other devices behave the same.**

I believe my next step in getting it running is to acquire a card that uses the AR9280 (officially supported by this project) or MT7610U (https://github.com/seemoo-lab/owl/issues/10) chipset. I'd prefer a USB card but couldn't find any that has the AR9280.

**Comment by thomasst** [comment]
@jackforfaltu I actually got it to work on AWUS036NHA. You'll need to put it in regular monitor mode and start owl with `-N` on channel 6, and also compile it using the patch from https://github.com/seemoo-lab/owl/pull/39. Then, if you put `opendrop` into receive mode, it should show up on an Apple device in Airdrop after waiting for around 10 seconds.

```
ifconfig yourwifi down
iw yourwifi set monitor control
sudo /path/to/owl/build/daemon/owl -N -i yourwifi -c 6
```

**Comment by CaptainStabs** [comment]
> If I start `owl` with `-N`, `opendrop find` seems to list the devices, but the device doesn't show up on my Mac or iPhone in AirDrop even though I have `opendrop receive` running (the device itself also shows up in `opendrop find`).

I could not get the recieve function to ever work properly. IIRC, you'll want to run the following commands:
`airmon-ng check kill`
`airmon-ng start wlan1`   the name may be different, use `lspci`  or `iw dev`. I don't remember which one.
`owl -i wlan1mon -N`      again, name may be different.
Then `opendrop receive` in a new terminal.

**Comment by CaptainStabs** [comment]
> Doesn't do 5 GHz though.
>  I'm assuming this is because 5 GHz is not supported

This should not matter, as I'm fairly certain only the "latest" iphone has 5g capability? 
You may also want a BLE device as well, as I'm fairly certain one of the forks of this has BLE support implented

**Comment by thomasst** [comment]
> I could not get the recieve function to ever work properly.

I only use it for receiving. It works alright on the AWUS036NHA if you apply the patch from https://github.com/seemoo-lab/owl/pull/39 but there is still some packet loss.


> This should not matter, as I'm fairly certain only the "latest" iphone has 5g capability?

Huh, I'm not talking about 5G. I'm talking about 5 GHz wi-fi (channels 44 and 149), which the AWUS036NHA doesn't support -- but it works on channel 6.


> I'm fairly certain one of the forks of this has BLE support implented

This one? https://github.com/llamakko/opendrop2 claims it's compatible with https://github.com/hexway/apple_bleee

I haven't looked into this though. I think that's needed to discover devices so you can send, not for receiving.

**Comment by CaptainStabs** [comment]
> Huh, I'm not talking about 5G. I'm talking about 5 GHz 

Yeah that's my mistake. Shouldn't respond to issues when sleep deprived

**Comment by SoniaMalki** [comment]
Hey ! I searched for the active monitor mode a lot, and I tried to find an adapter that support it. I found a GitHub that is very well maintained and I discussed the issue with the owner. Here is a link for the discussion that can help you find an adapter that will work for you. 
The owner told me which chipset works with active monitor mode, and if you grab this info you can search on the README of his GitHub an adapter with the mentioned chipset that will work for you:  https://github.com/morrownr/USB-WiFi/issues/73

thanks to him, we now have a very modern device list that will work with OWL.

**Comment by lucasromeiro** [comment]
> Hey ! I searched for the active monitor mode a lot, and I tried to find an adapter that support it. I found a GitHub that is very well maintained and I discussed the issue with the owner. Here is a link for the discussion that can help you find an adapter that will work for you. The owner told me which chipset works with active monitor mode, and if you grab this info you can search on the README of his GitHub an adapter with the mentioned chipset that will work for you: [morrownr/USB-WiFi#73](https://github.com/morrownr/USB-WiFi/issues/73)
> 
> thanks to him, we now have a very modern device list that will work with OWL.

@SoniaMalki 
Hello, thanks for the tip.

Could you give us more information?

Did your test work and was it successful with the file transfer?

I didn't find your feedback on github.

Thanks!
